common/djangoapps/third_party_auth/models.py

```diff
@@ -76,6 +78,131 @@ def __str__(self):
         )
 
 
+class UserDataSharingConsentAudit(TimeStampedModel):
+    """
+    Object that exists to store the canonical state of whether a particular
+    user has given consent for their course data to be shared with a particular
+    single-sign-on identity provider
+    """
+    class Meta(object):
+        app_label = 'third_party_auth'
+        verbose_name = "Data Sharing Consent Audit State"
+        verbose_name_plural = "Data Sharing Consent Audit States"
+
+    NOT_SET = 'not_set'
+    ENABLED = 'enabled'
+    DISABLED = 'disabled'
+    STATE_CHOICES = (
+        (NOT_SET, 'Not set'),
+        (ENABLED, 'Permitted'),
```

Comment:
Let's stay consistent and use 'Enabled' and 'Disabled' for the strings here, instead of 'Permitted' and 'Not permitted'.